File: G-for-g-Practice/Unique Substrings.java


// User function Template for Java

class Solution {
    public static int unique_substring(String s) {
        // Your code here
        Set <String> uniqueSs = new HashSet<>();
        for (int i=0; i<s.length(); i++){
            for(int j= i+1; j<=s.length(); j++){
                uniqueSs.add(s.substring(i,j));
            }
        }
        return uniqueSs.size();
    }
}

File: G-for-g-Practice/Unique Substrings.py
#User function Template for python3

class Solution:
    def unique_substring(self,s):
        #code here
        uniqueSs =set()
        for i in range (len(s)):
            for j in range (i+1, len(s)+1):
                uniqueSs.add(s[i:j])
        return len(uniqueSs)


